src/coreclr/tools/dotnet-pgo/Program.cs

```diff
@@ -1343,6 +1344,11 @@ private int InnerProcessTraceFileMain()
 
                 double excludeEventsBefore = Get(_command.ExcludeEventsBefore);
                 double excludeEventsAfter = Get(_command.ExcludeEventsAfter);
+                Regex excludeEventsBeforeMethod = !string.IsNullOrEmpty(Get(_command.ExcludeEventsBeforeMethod)) ? new Regex(Get(_command.ExcludeEventsBeforeMethod)) : null;
+                Regex excludeEventsAfterMethod = !string.IsNullOrEmpty(Get(_command.ExcludeEventsAfterMethod)) ? new Regex(Get(_command.ExcludeEventsAfterMethod)) : null;
+
+                // Find all the R2RLoad events.
+                SortedDictionary<double, List<Tuple<int, MethodDesc>>> sortedR2RLoadEvents = new();
```

Comment:
Can you please switch this to C# value tuples instead of `Tuple`?

Reply:
Also, do we need to sort these events on our own? Events in the trace should already be causally ordered.

Reply:
nettrace file will be relative sorted based on sequence points included within the file but between two sequence point there is no sort order in net trace v4 format, https://github.com/microsoft/perfview/blob/main/src/TraceEvent/EventPipe/EventPipeFormat.md#event-timestamp-sorting. Since the existing solution using --exclude-events-before, --exclude-events-after didn't just break out of the loop when detecting events that had a timestamp after --exclude-events-after I assumed code already had the assumption the the events might not be fully sorted on timestamp.

Reply:
I wasn't aware of this detail. However, dotnet-pgo uses TraceEvent under the hood and converts all input formats to .etlx files before processing the trace. I went and checked, and this format always has sorted events, so TraceEvent does the sort for us ([here](https://github.com/microsoft/perfview/blob/766a08ca8373d273d55da8b07147e7068e4d5f58/src/TraceEvent/EventPipe/EventCache.cs#L169)), so I don't think we need a manual sort here.